lib/fulfil/response_parser.rb

```diff
@@ -55,6 +55,9 @@ def self.group(key_value_tuples)
     end
 
     def self.parse(item:)
+      # If the item is not a hash, then it's a value and we can return it.
+      return item unless item.is_a?(Hash)
+
       key_value_tuples = item.to_a.map { |item_tuple| [item_tuple[0].split('.'), item_tuple[1]] }
```

Comment:
Can we refactor this a bit to;

```suggestion
      case item
      when Hash
        group(item.map({ |(key, value)| [key.split('.'), value] })).to_h
      else
        item
      end
```

Reply:
Sure